index.js

```diff
@@ -102,6 +104,57 @@ function fastifyPostgres (fastify, options, next) {
     }
   }
 
+  fastify.addHook('onRoute', routeOptions => {
+    const useTransaction = routeOptions.useTransaction
+
+    const transactionFailedSymbol = Symbol('transactionFailed')
+
+    if (useTransaction) {
+      // This will rollback the transaction if the handler fails at some point
+      const onError = async (req, reply, error) => {
+        req[transactionFailedSymbol] = true
+
+        try {
+          await req.pg.query('ROLLBACK')
+        } catch (err) {
+          // await req.pg.query('ROLLBACK')
+        }
+      }
+
+      const preHandler = async (req, reply) => {
+        const client = await pool.connect()
+        req.pg = client
+        await req.pg.query('BEGIN')
```

Comment:
awaiting at the end of an async function is an anti pattern https://www.nearform.com/blog/javascript-promises-the-definitive-guide/